assets/js/base/context/cart-checkout/event-emit/emitters.js

```diff
@@ -25,11 +25,16 @@ const getObserversByPriority = ( observers, eventType ) => {
 export const emitEvent = async ( observers, eventType, data ) => {
 	const observersByType = getObserversByPriority( observers, eventType );
 	let emitterResponse = {};
+	const observerResponses = [];
 	for ( const observer of observersByType ) {
 		try {
-			const response = await Promise.resolve( observer.callback( data ) );
-			if ( typeof response === 'object' ) {
-				emitterResponse = { ...emitterResponse, ...response };
+			const observerResponse = await Promise.resolve(
+				observer.callback( data )
+			);
+			if ( typeof observerResponse === 'object' ) {
+				emitterResponse = { ...emitterResponse, ...observerResponse };
+			} else {
+				observerResponses.push( observerResponse );
 			}
```

Comment:
I think if observerResponse is not an object, we'd have to assume it's okay and just not push to the collected array. There is no aborting so an observer returning anything other than an object has nothing useful to signal to the emitter right? So I don't think status changes should happen if there's nothing useful to indicate _why_ status is changing.